projects/views.py

```diff
@@ -1,37 +1,95 @@
 import json
 import random
 
+from django.core.serializers.json import DjangoJSONEncoder
+from django.http import HttpResponseRedirect
 from django.shortcuts import render
-from django.urls import reverse_lazy
-from django.views.generic.edit import CreateView, UpdateView
+from django.urls import reverse
 from django.views.generic.list import ListView
 
-from .forms import ProjectForm
+from projects.models import ProjectPhase, ProjectType
+
+from .forms import create_section_form_class
 from .models import Attribute, Project
 
 
-class ProjectCreateView(CreateView):
+class ProjectListView(ListView):
     model = Project
-    form_class = ProjectForm
-    template_name = 'project_form.html'
-    success_url = reverse_lazy('project-list')
+    template_name = 'project_list.html'
 
 
-class ProjectUpdateView(UpdateView):
-    model = Project
-    form_class = ProjectForm
-    template_name = 'project_form.html'
-    success_url = reverse_lazy('project-list')
+def generate_sections(project: Project=None):
+    if project and project.phase:
+        phase = project.phase
+    else:
+        phase = ProjectPhase.objects.get(index=0)
 
-    def get_initial(self):
-        initial = super().get_initial()
-        initial.update(self.object.attribute_data)
-        return initial
+    sections = []
+    for section in phase.sections.order_by('index'):
+        section_data = {
+            'section': section,
+            'form_class': create_section_form_class(section),
+            'form': None,
+        }
 
+        sections.append(section_data)
 
-class ProjectListView(ListView):
-    model = Project
-    template_name = 'project_list.html'
+    return sections
+
+
+def filter_data(identifiers, data):
+    filtered_data = {
+        key: value
+        for key, value in data.items()
+        if key in identifiers
+    }
+
+    return json.loads(json.dumps(filtered_data, cls=DjangoJSONEncoder))
+
+
+def project_edit(request, pk=None):
+    if pk:
+        project = Project.objects.get(pk=pk)
+    else:
+        project = Project()
+        project.phase = ProjectPhase.objects.get(index=0)
```

Comment:
The same as in the attribute importer, and actually more important here; While we don't really use project types yet, imo this should also take the project type into consideration to prevent quite cryptic failing in case someone has created phases for multiple project types.